include/nbl/system/demote_promote_writer_readers_lock.h

```diff
@@ -1,64 +1,88 @@
 #ifndef __NBL_DEMOTE_PROMOTE_WRITER_READERS_LOCK_H_INCLUDED__
 #define __NBL_DEMOTE_PROMOTE_WRITER_READERS_LOCK_H_INCLUDED__
 
-#include <boost/atomic/atomic.hpp>
 #include <thread>
-#include <mutex> // for std::adopt_lock_t
 
-// TODO: Reset spins if have to yield?
-// TODO atomic_unsigned_lock_free? since C++20 (from SReadWriteSpinLock)
+// TODO: Bring back proper memory semantics on fetch/store
 
 namespace nbl::system
 {
 
-class demote_promote_writer_readers_lock
+// Ugly workaround, structs with constevals can't be nested
+namespace impl
 {
-	constexpr static inline uint32_t MaxActors = 1023;
-	constexpr static inline uint32_t SpinsBeforeYield = 5000;
-	struct StateSemantics
+	template<typename U>
+	struct DPWRLstate_lock
+	{
+		using type = std::atomic<U>;
+	};
+
+	// will fail if `atomic_unsiged_lock_free` doesn't exist or its value type has wrong size
```

Comment:
btw another solution is to have `constexpr` function like `flipLockMaks` instead of the ``flipLock` `constexpr static inline` variable